sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala

```diff
@@ -656,8 +656,173 @@ class ForgerStakeMsgProcessorTest
     }
   }
 
+  @Test
+  def testRejectSendingPositiveValueToWithdraw(): Unit = {
+    val expectedBlockSignerProposition = "aa22334455667788112233445586778811223344556677881122334455667788" // 32 bytes
+    val blockSignerProposition = new PublicKey25519Proposition(BytesUtils.fromHexString(expectedBlockSignerProposition)) // 32 bytes
+    val expectedVrfKey = "aabbccddeeff0099aabb87ddeeff0099aabbccddeeff0099aabbccd2aeff001234"
+    val vrfPublicKey = new VrfPublicKey(BytesUtils.fromHexString(expectedVrfKey)) // 33 bytes
+
+    Mockito.when(mockNetworkParams.restrictForgers).thenReturn(true)
+    Mockito.when(mockNetworkParams.allowedForgersList).thenReturn(Seq(
+      (blockSignerProposition, vrfPublicKey)
+    ))
+
+    usingView(forgerStakeMessageProcessor) { view =>
+
+      forgerStakeMessageProcessor.init(view)
+
+      // create sender account with some fund in it
+      // val initialAmount = BigInteger.valueOf(10).multiply(validWeiAmount)
+      val initialAmount = ZenWeiConverter.MAX_MONEY_IN_WEI
+      createSenderAccount(view, initialAmount)
+
+      val cmdInput = AddNewStakeCmdInput(
+        ForgerPublicKeys(blockSignerProposition, vrfPublicKey),
+        ownerAddressProposition
+      )
+      val addNewStakeData: Array[Byte] = cmdInput.encode()
+
+      val stakeAmount = validWeiAmount
+      val addNewStakeMsg = getMessage(contractAddress, stakeAmount, BytesUtils.fromHexString(AddNewStakeCmd) ++ addNewStakeData, randomNonce)
+      val expStakeId = forgerStakeMessageProcessor.getStakeId(addNewStakeMsg)
+      val forgingStakeInfo = AccountForgingStakeInfo(expStakeId, ForgerStakeData(ForgerPublicKeys(blockSignerProposition, vrfPublicKey), ownerAddressProposition, stakeAmount))
+      val addNewStakeReturnData = withGas(forgerStakeMessageProcessor.process(addNewStakeMsg, view, _, defaultBlockContext))
+      assertNotNull(addNewStakeReturnData)
+
+
+      val nonce = randomNonce
+      val msgToSign = ForgerStakeMsgProcessor.getMessageToSign(forgingStakeInfo.stakeId, origin, nonce.toByteArray)
+      val msgSignatureData = Sign.signMessage(msgToSign, pair, true)
+      val msgSignature = new SignatureSecp256k1(msgSignatureData)
+
+      // create command arguments
+      val removeCmdInput = RemoveStakeCmdInput(forgingStakeInfo.stakeId, msgSignature)
+      val data: Array[Byte] = removeCmdInput.encode()
+      val msg = getMessage(contractAddress, BigInteger.ONE, BytesUtils.fromHexString(RemoveStakeCmd) ++ data, nonce)
+
+      // try processing the removal of stake, should succeed
+      val returnData = intercept[ExecutionRevertedException] {
+        withGas(forgerStakeMessageProcessor.process(msg, view, _, defaultBlockContext))
+      }
+
+      assert(returnData.getMessage === "Call value must be zero")
+
+      view.commit(bytesToVersion(getVersion.data()))
+    }
+  }
+
+  @Test
+  def testRejectSendingNegativeValueToWithdraw(): Unit = {
```

Comment:
Don't duplicate 50 lines of code just to change the sign of value.

Reply:
I've seen this in other places too, but why do we even need for test negative values? It is not possible to create a transaction with negative value, isn't it?

Reply:
This line was calling view method with negative value before https://github.com/HorizenOfficial/Sidechains-SDK/blob/dev_evm/sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala#L296 so I wasn't sure if negatives were allowed. Before the change, this test was passing.